src/core/Akka.Streams.Tests/Dsl/KeepAliveConcatSpec.cs

```diff
@@ -143,15 +143,19 @@ public void KeepAliveConcat_should_emit_buffered_elements_when_upstream_complete
                 .Via(new KeepAliveConcat<int>(5, TimeSpan.FromSeconds(60), x => new[] { x }))
                 .RunWith(Sink.FromSubscriber(downstream), Sys.Materializer());
 
-            upstream.SendNext(1);
-            upstream.SendNext(2);            
-            upstream.SendComplete();
+            // Wait for KeepAliveConcat to pull from upstream (respecting reactive streams protocol)
+            await upstream.ExpectRequestAsync();
```

Comment:
This is the key fix - `SendNext` will not work if demand is not detected on the stream.